synapse/http/client.py

```diff
@@ -64,13 +66,23 @@ class SimpleHttpClient(object):
     """
     def __init__(self, hs):
         self.hs = hs
+
+        pool = HTTPConnectionPool(reactor)
+
+        # the pusher makes lots of concurrent SSL connections to sygnal, and
+        # tends to do so in batches, so we need to allow the pool to keep lots
+        # of idle connections around.
+        pool.maxPersistentPerHost = max((100 * CACHE_SIZE_FACTOR, 5))
```

Comment:
Although a 100 connections per host seems a bit overkill for a pool to have tbh. I'm not sure whether its worth making this configurable via CACHE_SIZE_FACTOR to be honest.

Reply:
I guess the point is that the pusher is a bit silly and needs that many connections, and so we may as well keep them around?

Reply:
the point is as described in the comments: yes it needs that many connections, and if we don't keep them around we end up killing ourselves with ssl nego.

using CACHE_SIZE_FACTOR was a crude hack to make it pool 1000 connections on matrix.org, and fewer on smaller homeservers.